### ISSUE TITLE
[BUG] - Client build freezes when connecting to server because of Pixel Streaming arguments 

### ISSUE THREAD

**Issue by kaelyavel** [EpicGames/PixelStreamingInfrastructure-archived-2024#330]
**UE Version: **
E.g. UE 5.2.1 (build from source)

**Frontend Version: **
One on the UE5.2 branch

**Problem component**

Pixel Streaming C++ plugin

**Description**
I packaged the Third Person Template project with very minor changes to generate a Client build and a dedicated server (I built the engine from source to have the solution configurations)

My aim is to do Pixel Streaming x Multiplayer, in other words, Pixel Stream to a web page (Matchmaker then Signalling Web Server) a Client build connected to a dedicated server, and I succeeded at doing that, but I encountered many times a really annoying bug that freezes my Client build when it connects to the dedicated server through 'open {dedicated_server_ip_address}'.

Just to lay out my architecture : 

My dedicated server sits on a AWS instance that has Windows Server 2022 Base and no GPU. (a t2.medium : 2vCPUS and 4Gb of RAM) along with the Pixel Streaming Matchmaker.

My Client build sits on a AWS instance that has Windows Server 2019 and a GPU (a g4dn.xlarge : 4vCPUs, 16GB of RAM, and a Nvidia T4 GPU) along with the Signalling Web Server.

I made sure to open ports 80, 8888 for the SWS and ports 90, 9999 for the Matchmaker, in addition to port 7777 for the dedicated server.

Even though I'm citing my architecture, I deduced after many tests that the problem is independent from that.

When I connect my Client build to the dedicated server with no _"-PixelStreamingIP=localhost -PixelStreamingPort=8888"_ as startup arguments on the Client, the connection succeeds at every try, but as soon as I add  _"-PixelStreamingIP=localhost -PixelStreamingPort=8888"_ as startup arguments to the client and despite having no SWS running aside, I start experiencing endless freezes on it each time I use 'open {dedicated_server_ip_address}' to connect it to the dedicated server.

When I checked the dedicated server during the client freeze, I found out there was a first connection with the client but it warned  concerning the delay between received packets before closing the connection.

I have done a large amount of varied tests and I can assure that a connection to a dedicated server with 'open {dedicated_server_ip_address}' on a Client build launched with Pixel Streaming arguments leads to freezes on it.
Please refer to the video I attached below.

**Steps to Reproduce:**
1. Build Client and dedicated server of the Third Person template project 
2. Put Client build and dedicated server on different machines
3. Open ports mentioned above
4. Launch Dedicated server (With -log as startup argument preferably)
5. Launch Client build with "-PixelStreamingIP=localhost -PixelStreamingPort=8888"_ as startup arguments
6. Press on ~ to open UE console command
7. Connect Client build to dedicated server with 'open {dedicated_server_ip_address}'
8. You'll notice that the Client build Freezes indefinitely. 

**Expected behavior**
1. Client build connected to dedicated server with no freeze

**Video**
What is presented in this video is the Client build on a AWS instance trying to connect to a dedicated server on another AWS instance.
SWS is running in this video but the freeze happen even when it's not.

https://github.com/EpicGames/PixelStreamingInfrastructure/assets/57868366/fb1591af-e4b9-4954-88b9-37e35baff6c3

**Comment by PsiCondition** [comment]
Try https://arcware.com/cloudrt for your clients !
We have successfully achieved Multiplayer on our platform !

AWS is history !

Sent from my iPhone

On 26. Jul 2023, at 01:32, El Bachir ***@***.***> wrote:

﻿

**UE Version: **
E.g. UE 5.2.1 (build from source)

**Frontend Version: **
One on the UE5.2 branch

Problem component

Pixel Streaming C++ plugin

Description
I packaged the Third Person Template project with very minor changes to generate a Client build and a dedicated server (I built the engine from source to have the solution configurations)

My aim is to do Pixel Streaming x Multiplayer, in other words, Pixel Stream to a web page (Matchmaker then Signalling Web Server) a Client build connected to a dedicated server, and I succeeded at doing that, but I encountered many times a really annoying bug that freezes my Client build when it connects to the dedicated server through 'open {dedicated_server_ip_address}'.

Just to lay out my architecture :

My dedicated server sits on a AWS instance that has Windows Server 2022 Base and no GPU. (a t2.medium : 2vCPUS and 4Gb of RAM) along with the Pixel Streaming Matchmaker.

My Client build sits on a AWS instance that has Windows Server 2019 and a GPU (a g4dn.xlarge : 4vCPUs, 16GB of RAM, and a Nvidia T4 GPU) along with the Signalling Web Server.

I made sure to open ports 80, 8888 for the SWS and ports 90, 9999 for the Matchmaker, in addition to port 7777 for the dedicated server.

Even though I'm citing my architecture, I deduced after many tests that the problem is independent from that.

When I connect my Client build to the dedicated server with no "-PixelStreamingIP=localhost -PixelStreamingPort=8888" as startup arguments on the Client, the connection succeeds at every try, but as soon as I add "-PixelStreamingIP=localhost -PixelStreamingPort=8888" as startup arguments to the client and despite having no SWS running aside, I start experiencing endless freezes on it each time I use 'open {dedicated_server_ip_address}' to connect it to the dedicated server.

When I checked the dedicated server during the client freeze, I found out there was a first connection with the client but it warned concerning the delay between received packets before closing the connection.

I have done a large amount of varied tests and I can assure that a connection to a dedicated server with 'open {dedicated_server_ip_address}' on a Client build launched with Pixel Streaming arguments leads to freezes on it.
Please refer to the video I attached below.

Steps to Reproduce:

  1.  Build Client and dedicated server of the Third Person template project
  2.  Put Client build and dedicated server on different machines
  3.  Open ports mentioned above
  4.  Launch Dedicated server (With -log as startup argument preferably)
  5.  Launch Client build with "-PixelStreamingIP=localhost -PixelStreamingPort=8888"_ as startup arguments
  6.  Press on ~ to open UE console command
  7.  Connect Client build to dedicated server with 'open {dedicated_server_ip_address}'
  8.  You'll notice that the Client build Freezes indefinitely.

Expected behavior

  1.  Client build connected to dedicated server with no freeze

Video
SWS is launched in this video but the freeze happen even when it's not.

https://github.com/EpicGames/PixelStreamingInfrastructure/assets/57868366/fb1591af-e4b9-4954-88b9-37e35baff6c3

**Comment by kaelyavel** [comment]
I solved this issue.

It turn out it was the AWS AMI (Image) that I was using to launch my instances. I was using the [NICE DCV for Windows (g4 and g5 with NVIDIA gaming driver)](https://aws.amazon.com/marketplace/pp/prodview-67uoddn7zjdxu?sr=0-2&ref_=beagle&applicationId=AWSMPContessa) that comes with GPU drivers installed for g4dn (T4) instances and some streaming protocoles. When I launched a non-AMI g4dn instance, installed the GPU drivers, launched my clients builds and connected to the dedicated server (on another instance), those freezes never happened.

I'm keeping this issue opened for the moment before closing it in some days (or repository staff doing it). 

Keep up the great work.
